Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -288,13 +479,65 @@ namespace GradientSignal
                 // Flip the y because images are stored in reverse of our world axes
                 y = (height - 1) - y;
 
-                return AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y);
+                // By default, we will query from the R channel of the image
+                auto channel = ChannelToUse::R;
+
+                // Advanced handling of channel and scale
+                if (m_configuration.m_advancedMode)
+                {
+                    channel = m_configuration.m_channelToUse;
+
+                    // For terrarium, there is a separate algorithm for retrieving the value
+                    if (channel == ChannelToUse::Terrarium)
+                    {
+                        return GetTerrariumPixelValue(x, y);
+                    }
+
+                    // Handle custom user scale
+                    if (m_configuration.m_customScaleType == CustomScaleType::Manual)
+                    {
+                        float value = AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(channel));
+                        auto formatRange = GetPixelFormatRange(imageDescriptor.m_format);
+                        float scaleFactor = formatRange.second - formatRange.first;
+                        return value * ((m_configuration.m_scaleRangeMax / scaleFactor) - (m_configuration.m_scaleRangeMin / scaleFactor)) + (m_configuration.m_scaleRangeMin / scaleFactor);
+                    }
+                }
+
+                return AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(channel));
             }
         }
 
         return defaultValue;
     }
 
+    float ImageGradientComponent::GetTerrariumPixelValue(AZ::u32 x, AZ::u32 y) const
+    {
+        const AZ::RHI::ImageDescriptor& imageDescriptor = m_configuration.m_imageAsset->GetImageDescriptor();
+
+        float r = AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(ChannelToUse::R));
+        float g = AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(ChannelToUse::G));
+        float b = AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(ChannelToUse::B));
+
+        // Convert back to the unsigned 8-bit value
+        float maxValue = aznumeric_cast<float>(std::numeric_limits<AZ::u8>::max());
+        r *= maxValue;
+        g *= maxValue;
+        b *= maxValue;
```

Comment:
Just to check, are you absolutely sure that GetImageDataPixelValue is always going to return a 0-1 value here?

Reply:
In this case, yes since we have an additional restriction on supported formats for using terrarium `DoesFormatSupportTerrarium`